internal/clients/elasticsearch/watch.go

```diff
@@ -0,0 +1,84 @@
+package elasticsearch
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"net/http"
+
+	"github.com/elastic/terraform-provider-elasticstack/internal/clients"
+	"github.com/elastic/terraform-provider-elasticstack/internal/models"
+	"github.com/elastic/terraform-provider-elasticstack/internal/utils"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+)
+
+func PutWatch(ctx context.Context, apiClient *clients.ApiClient, watch *models.Watch) diag.Diagnostics {
+	var diags diag.Diagnostics
+	watchBodyBytes, err := json.Marshal(watch.Body)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	esClient, err := apiClient.GetESClient()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	body := esClient.Watcher.PutWatch.WithBody(bytes.NewReader(watchBodyBytes))
+	active := esClient.Watcher.PutWatch.WithActive(watch.Status.State.Active)
```

Comment:
Re-using the response model feels a little confusing, I think it would be nicer to have a `models.PutWatch` struct with a top level `Active` property.

Reply:
Nice, didn't think of this! Hoping this is what you mean: [b4e9bcf](https://github.com/elastic/terraform-provider-elasticstack/pull/155/commits/b4e9bcfa03df184720fc82bc8edf11239b663242)